File: Rest.java
//
// To run this integrations use:
//
//     kamel run --name=restjava --dependency=camel-restlet  Rest.java
//
//
//import org.springframework.stereotype.Component;
import org.apache.camel.builder.RouteBuilder;

//@Component
public class Rest extends RouteBuilder {
    @Override
    public void configure() throws Exception {
        restConfiguration()
            .component("restlet")
            .host("localhost")
            .port("8080");

        rest()
            .get("/hello")
            .to("direct:hello");

        from("direct:hello")
            .transform().simple("Hello World");
    }
}

File: rest.js
//
// To run this integrations use:
//
//     kamel run --name=withrestjs --dependency=camel-undertow rest.js
//

// ****************
//
// Setup
//
// ****************

l = components.get('log')
l.exchangeFormatter = function(e) {
    return "log - body=" + e.in.body + ", headers=" + e.in.headers
}

c = restConfiguration()
c.component = 'undertow'
c.port = 8080

// ****************
//
// Functions
//
// ****************

function proc(e) {
    e.getIn().setHeader('RandomValue', Math.floor((Math.random() * 100) + 1))
}

// ****************
//
// Route
//
// ****************

rest()
    //.path('/say/hello')
    .path('/')
    .produces("text/plain")
    .get().route()
        .to("direct:hello");

from("direct:hello")
        .transform().constant("Hello World");
        //.log('{body}');
/*
from('timer:js?period=1s')
    .routeId('js')
    .setBody()
        .constant('Hello Camel K')
    .process(proc)
    .to('log:info')
*/
